### ISSUE TITLE
Wifite on Kali Linux inside windows 10 can't detect any wireless interface

### ISSUE THREAD

**Issue by Reforge-Mark** [derv82/wifite2#231]
Have installed Kali Linux only for this purpose, but it fails to work.. 
`   .               .
 .´  ·  .     .  ·  `.  wifite 2.2.5
 :  :  :  (¯)  :  :  :  automated wireless auditor
 `.  ·  ` /¯\ ´  ·  .´  https://github.com/derv82/wifite2
   `     /¯¯¯\     ´

 [!] Warning: Recommended app hcxpcaptool was not found. install @ https://github.com/ZerBea/hcxtools

 [+] Checking airmon-ng...
 [!] airmon-ng did not find any wireless interfaces
 [!] Make sure your wireless device is connected
 [!] See http://www.aircrack-ng.org/doku.php?id=airmon-ng for more info

 [!] Error: airmon-ng did not find any wireless interfaces

 [!] Full stack trace below

 [!]    Traceback (most recent call last):
 [!]    File "/usr/lib/python2.7/dist-packages/wifite/__main__.py", line 93, in entry_point
 [!]        wifite.start()
 [!]    File "/usr/lib/python2.7/dist-packages/wifite/__main__.py", line 53, in start
 [!]        Configuration.get_monitor_mode_interface()
 [!]    File "/usr/lib/python2.7/dist-packages/wifite/config.py", line 129, in get_monitor_mode_interface
 [!]        cls.interface = Airmon.ask()
 [!]    File "/usr/lib/python2.7/dist-packages/wifite/tools/airmon.py", line 299, in ask
 [!]        raise Exception('airmon-ng did not find any wireless interfaces')
 [!]  Exception: airmon-ng did not find any wireless interfaces

 [!] Exiting` and yes I have a wi fi antenna which i'm using to catch any internet at all

**Comment by Reforge-Mark** [comment]
I've seen some similar issue with advice to dual boot linux and then try detecting the wireless interface.. but considering it took me almost a whole week to download this kali and the required programs for this wifite i'd really not want to have to download again a whole OS and then again whole bunch of other stuff again.. I live in a studio that is a part of a family house, they changed the password for wifi and are black mailing me to pay up much more than usual for internet.  I had to move whole PC, to some cranked corner next to my fridge to catch some weak signal that gets me like 1kb/s - 10kb/s speed and often breaks and goes below 1kb/s too.. Not to mention whole PC overheats (it's right next to the fridge and the wall) and has to be shutdown occasionally so any downloads are interrupted and stopped again. You can imagine my frustrations these days in trying to hack the password and get my normal net back.  Have seen some metasploit framework that works on windows too, but will probably take me 5-6 days to download it. In the meantime my private server project bills are incoming and I have to solve these problems to work and pay it. 

-----
Basically am looking into making this Wifite work through Kali linux on windows 10 through any means.

**Comment by kimocoder** [comment]
Does Windows subsyatem support I/O controllers e.g USB devices yet then? Did not last time I checked..

**Comment by Reforge-Mark** [comment]
That feeling when you work on something for over a week only to realize it does not even work .. guess i could try dual booting some ubuntu cd i have from 10 years ago or more.. pretty sure the cd is not working properly and even if installed probably would need so many updates and dependencies downloaded for wifite to function that it would take less days for some other solution.

**Comment by RaduNico** [comment]
As far as I can tell the Windows Subsystem for Linux won't work with wifite. I tried using a linux virtual machine and give direct access for the USB device but I had no luck in getting it working. Your best chance is creating a dual boot. A 10 year old ubuntu will most likely not be able to help you, even if the cd would be working properly - that would be a very old ubutu distribution and a lot more issues would arise.

Note that you would need an external USB WiFi dongle if your network card does not support monitor mode and there is a good chance it does not support it. You could check here [1] provided you know what chipset you have. On top of that wifite still depends on what dictionaries you have. If the password is not one in those 4800 passwords included in the repository you would need another dictionary and/or hashcat/john rules.

Your issue would be a bit more easy (and legal) to solve by talking to the owner. Relevant xkcd attached [2]. Or attach a cable to the router, hope it has default credentials and just get the password from it.

[1] - [https://wikidevi.com/wiki/Wireless_adapters/Chipset_table](https://wikidevi.com/wiki/Wireless_adapters/Chipset_table)
[2] - [https://www.xkcd.com/538/](https://www.xkcd.com/538/)

**Comment by Reforge-Mark** [comment]
Thank you RaduNico, but I have no access to any of their hardware or where to plug in a cable as I live in a studio separate from the family house.. and talking to father who does not talk to me unless asking for money or stepmother that abused me my entire childhood and broke my ribs since i was 10years old for fun certainly won't go the way you think it would. As the passwords for wifi are probably some words and numbers symbolic to them, it won't be simple to crack. I'll try next 7 days or more to download then dual boot from usb some linux and see if I can get somewhere then, if not will have to buy some much more powerful wi-fi antenna and catch some signals from further away that are not locked. Will close this issue now, as it seems wifite will not work under Windows Subsystem for Linux anytime soon and maybe there will never even be support for USB devices in such environment
